components/Schedule.js

```diff
@@ -0,0 +1,346 @@
+        flex-direction: row;
+    }
+`;
+
+const Col = styled.div`
+    flex 1;
+        
+`;
+
+const Slot = ({title, image, speaker, startTime, endTime, dark}) => {
+    let content;
+    let timeString = '';
+    if (startTime) {
+        timeString += startTime;
+    }
+    if (endTime) {
+        timeString += ` - ${endTime}`;
+    }
+    if (title) {
+        content = <Title image={image}>{title}</Title>;
+    } else if (speaker) {
+        let speaker2;
+        const {coSpeaker} = speaker;
+        if (coSpeaker) {
+            speaker2 = (
+                <SpeakerInfo2>
+                    <img src={coSpeaker.imageUrls.small} alt={coSpeaker.name} />
+                    <span>{coSpeaker.name}</span>
+                    <ImageOverlay2 />
+                </SpeakerInfo2>
+            );
+        }
+        content = (
+            <Speaker>
+                <img src={speaker.imageUrls.small} alt={speaker.name} />
+                <ImageOverlay />
+                <SpeakerInfo>
+                    {speaker.name}
+                    <div dangerouslySetInnerHTML={{__html: speaker.teaser}} />
+                </SpeakerInfo>
+                {speaker2}
+            </Speaker>
+        );
+    }
+
+    return (
+        <Block dark={dark}>
+            <TimeInfo>{timeString}</TimeInfo>
+            {content}
+        </Block>
+    );
+};
+
+Slot.propTypes = {
+    title: PropTypes.string,
+    image: PropTypes.string,
+    speaker: PropTypes.object,
+    startTime: PropTypes.string,
+    endTime: PropTypes.string,
+    dark: PropTypes.bool,
+};
+
+export default class Schedule extends React.PureComponent {
+    constructor(props) {
+        super(props);
+
+        this.state = {slots: this.parseSlots()};
+    }
+    static propTypes = {
+        speakers: PropTypes.array.isRequired,
+    };
+
+    parseSlots = () => {
+        const slots = new Array(15);
+        const {speakers} = this.props;
+
+        speakers.forEach(speaker => {
+            const slot = parseInt(speaker.slot);
+            if (slot) {
+                if (typeof slots[slot] === 'object') {
+                    const entry = slots[slot];
+                    entry.coSpeaker = speaker;
+                    slots[slot] = entry;
+                } else {
+                    slots[slot] = speaker;
+                }
+            }
+        });
+        return slots;
+    };
+
+    render() {
+        const {slots} = this.state;
+        return (
+            <div>
+                <Row>
+                    <Col>
+                        <Slot title="DAY 1 - Thursday, 25.01.2018" />
+                        <Slot dark={true} />
+                        <Slot
+                            title="Breakfast & Checkin"
+                            image="static/schedule/checkin.jpg"
+                            startTime="08:00"
+                            endTime="09:00" />
+                        <Slot speaker={slots[1]} dark={true} startTime="09:00" endTime="09:45" />
```

Comment:
I think I'd have the `startTime` and `endTime` included in the `slots`  object, so that I can just iterate over all the slots here... But I can see the advantage of not having to add the time in the CMS... So both variants are ok for me.

Reply:
decided to use the fixed way, not to change all the cms stuff.